synapse/storage/databases/main/registration.py

```diff
@@ -2312,6 +2312,37 @@ async def add_refresh_token_to_user(
 
         return next_id
 
+    def _move_device_refresh_token_txn(
+        self, txn: LoggingTransaction, old_device_id: str, device_id: str
+    ) -> None:
+        """Updates rows of old_device_id with current device_id"""
+
+        self.db_pool.simple_update_txn(
+            txn,
+            "refresh_tokens",
+            {"device_id": old_device_id},
+            {"device_id": device_id},
+        )
+
+    async def move_device_refresh_token(
+        self, old_device_id: str, device_id: str
+    ) -> None:
+        """Moves refresh tokens from old device to current device
+
+        Args:
+            old_device_id: The old device.
+            device_id: The new device ID.
+        Returns:
+            None
+        """
+
+        await self.db_pool.runInteraction(
+            "move_device_refresh_token",
+            self._move_device_refresh_token_txn,
+            old_device_id,
+            device_id,
+        )
```

Comment:
Instead of a custom function we can probably use the non-transaction version of `simple_update`:

```suggestion
        await self.db_pool.simple_update(
            ...
        )
```